jl4-core/src/L4/Evaluate/ValueLazy.hs

```diff
@@ -38,17 +38,25 @@ data NF = MkNF (Value NF) | ToDeep
   deriving anyclass NFData
 
 data Value a =
-    ValNumber Int -- for now
+    ValNumber Rational -- for now
   | ValString Text
   | ValNil
   | ValCons a a
   | ValClosure (GivenSig Resolved) (Expr Resolved) Environment
+  | ValUnaryBuiltinFun UnaryBuiltinFun
   | ValUnappliedConstructor Resolved
   | ValConstructor Resolved [a]
   | ValAssumed Resolved
   | ValEnvironment Environment
   deriving stock Show
 
+data UnaryBuiltinFun
+  = UnaryIsInteger
+  | UnaryRound
+  | UnaryCeiling
+  | UnaryFloor
+  deriving stock (Show)
+
```

Comment:
can we not reuse this datatype across lazy and strict evaluators? Perhaps not necessary since the strict one is being deprecated anyway, but it would still declutter things. Perhaps the strict value module could depend on the lazy value module.